src/context/WalletProvider/KeepKey/components/Pin.tsx

```diff
@@ -79,20 +83,41 @@ export const KeepKeyPin = () => {
     }
   }
 
+  const handleKeyboardInput = (e: KeyboardEvent) => {
+    // We can't allow tabbing between inputs or the focused element gets out of sync with the KeepKey
+    if (e.key === 'Tab') e.preventDefault()
+
+    if (e.key === 'Backspace') return
+
+    if (e.key === 'Enter') {
+      handleSubmit()
+      return
+    }
+
+    if (!pinNumbers.includes(Number(e.key))) {
+      e.preventDefault()
+      return
+    } else {
+      e.preventDefault()
+      handlePinPress(Number(e.key))
+      return
+    }
+  }
+
   // Use different translation text based on which type of PIN request we received
   const translationType = (() => {
     switch (keepKeyPinRequestType) {
       case PinMatrixRequestType.NEWFIRST:
         return 'newPin'
       case PinMatrixRequestType.NEWSECOND:
         return 'newPinConfirm'
-      default:
+      case PinMatrixRequestType.CURRENT:
```

Comment:
I was preferring the old default fallback because it's now polluting the markup with some defined checks while there are only 3 states possible and this component shouldn't be loaded at all if there are no pin request

Reply:
This was introducing the issue on the A/C 3: https://github.com/shapeshift/web/issues/1613#issue-1218137363

Reply:
But now, it will flash an empty title, button text or whatever?

It will be fixed by https://github.com/shapeshift/web/pull/2589 anyway as it will enter in a loading state

Reply:
Ok, will revert